osu.Game/Graphics/ScreenshotManager.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Drawing.Imaging;
+using System.IO;
+using osu.Framework.Allocation;
+using osu.Framework.Configuration;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Input;
+using osu.Framework.Input.Bindings;
+using osu.Framework.Platform;
+using osu.Game.Configuration;
+using osu.Game.Input.Bindings;
+
+namespace osu.Game.Graphics
+{
+    public class ScreenshotManager : Container, IKeyBindingHandler<GlobalAction>, IHandleGlobalInput
+    {
+        private Bindable<ScreenshotFormat> screenshotFormat;
+        private GameHost host;
+        private Storage storage;
+
+        [BackgroundDependencyLoader]
+        private void load(GameHost host, OsuConfigManager config, Storage storage)
+        {
+            this.host = host;
+            this.storage = storage.GetStorageForDirectory(@"screenshots");
+
+            screenshotFormat = config.GetBindable<ScreenshotFormat>(OsuSetting.ScreenshotFormat);
+        }
+
+        public bool OnPressed(GlobalAction action)
+        {
+            switch (action)
+            {
+                case GlobalAction.TakeScreenshot:
+                    TakeScreenshot();
+                    return true;
+            }
+
+            return false;
+        }
+
+        public bool OnReleased(GlobalAction action) => false;
+
+        public void TakeScreenshot()
+        {
+            host.TakeScreenshot(screenshotBitmap =>
+            {
+                var stream = storage.GetStream($"{DateTime.Now:yyyyMMddTHHmmss}.{screenshotFormat.ToString().ToLower()}", FileAccess.Write);

```

Comment:
this isn't going to work. as discussed in discord, you should be checking whether the file already exists and add a unique suffix to the end of the filename.

currently crashes if you hit F12 too many times in a short period.

Reply:
Do you mean that it should be an integer stored in the game config?

Reply:
`original.png`
Check if file exists, if so, increment suffix.
`original-1.png`
Check if file exists, if so, increment suffix.
`original-2.png`
Doesn't exist, use.